### PR TITLE
PRO-1409 Upgrade Sign Modal

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1409/upgrade-sign-modal

## Description

- Upgraded sign modal for `eth_signTypedData` method

## How Has This Been Tested?

- Tested on Android Device

## Screenshots (if appropriate):

https://github.com/pillarwallet/pillarwallet/assets/82652040/12178785-2bb8-425d-9934-d72c061d7f1c

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by Yash-Chaubal — APPROVED** [review]
LGTM!

File: src/screens/WalletConnect/CallRequest/SignatureRequestContent.js [hunk 1]
```diff
@@ -55,11 +60,53 @@ function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
...
   const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
 
-  const { title, iconUrl, chain } = getViewData(request);
+  const { title, iconUrl, chain, method, params } = getViewData(request);
   const config = configs[chain];
 
+  const isSignTypedData = method === ETH_SIGN_TYPED_DATA || method === ETH_SIGN_TYPED_DATA_V4;
+
+  const { message, primaryType } =
+    isSignTypedData && !isEmpty(params) ? JSON.parse(params[1]) : { message: null, primaryType: null };
```
Comment (IAmKio) [inline src/screens/WalletConnect/CallRequest/SignatureRequestContent.js:72]: `JSON.parse(params[1])` will throw an error if that index doesn't exist. Could we check for this or fail gracefully?

File: src/screens/WalletConnect/CallRequest/SignatureRequestContent.js [hunk 2]
```diff
@@ -55,11 +60,53 @@ function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
...
+    const values = Object.entries(message);
+    if (isEmpty(values)) return [];
+
+    return values.map((value) => {
+      const key = value?.[0];
+      const description = value?.[1];
+
+      if (!key) return '';
+
+      const label = key.charAt(0).toUpperCase() + key.slice(1);
```
Comment (IAmKio) [inline src/screens/WalletConnect/CallRequest/SignatureRequestContent.js:86]: if `label` is a `string`, then use string concatenation via template strings instead:

```suggestion
      const label = `${key.charAt(0).toUpperCase()}${key.slice(1)}`;
```

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few small points but other than that it looks good

File: src/screens/WalletConnect/CallRequest/SignatureRequestContent.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/screens/WalletConnect/CallRequest/SignatureRequestContent.js]: Condition updated.

File: src/screens/WalletConnect/CallRequest/SignatureRequestContent.js (hunk 2 above)
Comment (lalitkumarrock) [reply src/screens/WalletConnect/CallRequest/SignatureRequestContent.js]: Updated it!

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**